fix_includes_test.py

```diff
@@ -1884,9 +1885,76 @@ def testIncludeComments(self):
     self.RegisterFileContents({'subdir/include_comments.cc': infile})
     self.ProcessAndTest(iwyu_output)
 
+  def testIncludeNoComments(self):
+    """Tests that we don't include comments on #include lines. with --comments=none"""
+    self.flags.comments = 'none'
+    infile = """\
+// Copyright 2010
+
+#include <notused.h>  ///-
+///+#include "subdir/include_no_comments.h"
+///+#include <stdio.h>
+#include "used.h"
+///+#include "used2.h"
+
+int main() { return 0; }
+"""
+    iwyu_output = """\
+subdir/include_no_comments.cc should add these lines:
+#include "subdir/include_no_comments.h"
+#include <stdio.h>
+#include "used2.h"
+
+subdir/include_no_comments.cc should remove these lines:
+- #include <notused.h>  // lines 3-3
+
+The full include-list for subdir/include_no_comments.cc:
+#include "subdir/include_no_comments.h"
+#include <stdio.h>
+#include "used.h"
+#include "used2.h"
+---
+"""
+    self.RegisterFileContents({'subdir/include_no_comments.cc': infile})
+    self.ProcessAndTest(iwyu_output)
+
+  def testIncludeLongComments(self):
```

Comment:
Wait a minute. Sorry I'm asking this at this point, but it hadn't occurred to me before: does the value of `--comments` affect the processing in `fix_includes.py` at all?

It looks like it's now:
* `--comments=none` - strip comments from IWYU output
* `--comments=short`, `--comments=long` - use comments for append-lines from IWYU output
* `--update_comments=none` - strip comments if they exist, even if IWYU suggests no changes
* `--update_comments=short`, `--update_comments=long` - use comments from IWYU output, even if IWYU suggests no changes

Maybe we/I need to take a step back again and think about what the switches really mean.

Intuitively, this makes me think that the `fix_includes.py` behavior should not have needed patching in order to add longer comments? The original behavior was (correct me if I'm wrong):

* `--comments` - use comments for append-lines from IWYU output
* `--update_comments` - use comments for all include-lines from IWYU output

which seems simple enough...

Sorry about the late realization, hopefully I've overlooked something :-)